specification/communication/data-plane/NetworkTraversal/stable/2022-03-01/CommunicationNetworkTraversal.json

```diff
@@ -0,0 +1,179 @@
+          "type": "string"
+        },
+        "routeType": {
+          "description": "Filter the routing methodology returned. If not provided, will return all route types in separate ICE servers.",
+          "$ref": "#/definitions/RouteType"
+        },
+        "ttl": {
+          "type": "integer",
+          "format": "int32",
+          "description": "The credential Time-To-Live (TTL), in seconds. The default value (172800) will be used if given value exceeds it."
+        }
+      }
+    },
+    "CommunicationIceServer": {
+      "description": "An instance of a STUN/TURN server with credentials to be used for ICE negotiation.",
+      "required": [
+        "credential",
+        "urls",
+        "username",
+        "routeType"
+      ],
+      "type": "object",
+      "properties": {
+        "urls": {
+          "description": "List of STUN/TURN server URLs.",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "username": {
+          "description": "User account name which uniquely identifies the credentials.",
+          "type": "string"
+        },
+        "credential": {
+          "description": "Credential for the server.",
+          "type": "string"
+        },
+        "routeType": {
+          "$ref": "#/definitions/RouteType"
+        }
+      }
+    },
+    "CommunicationRelayConfiguration": {
+      "description": "A relay configuration containing the STUN/TURN URLs and credentials.",
+      "required": [
+        "expiresOn",
+        "iceServers"
+      ],
+      "type": "object",
+      "properties": {
+        "expiresOn": {
+          "format": "date-time",
+          "description": "The date for which the username and credentials are not longer valid. Will be 48 hours from request time.",
+          "type": "string"
+        },
+        "iceServers": {
+          "description": "An array representing the credentials and the STUN/TURN server URLs for use in ICE negotiations.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/CommunicationIceServer"
+          }
+        }
+      }
+    },
+    "RouteType": {
+      "description": "The routing methodology to where the ICE server will be located from the client. \"any\" will have higher reliability while \"nearest\" will have lower latency. It is recommended to default to use the \"any\" routing method unless there are specific scenarios which minimizing latency is critical.",
+      "enum": [
+        "any",
+        "nearest"
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "name": "RouteType",
+        "modelAsString": true
+      }
+    }
+  },
+  "parameters": {
+    "ApiVersionParameter": {
+      "in": "query",
+      "name": "api-version",
+      "description": "Version of API to invoke.",
+      "required": true,
+      "enum": [
+        "2022-03-01"
+      ],
```

Comment:
Please remove the enum from this definition or it will cause you perpetual grief.

Reply:
QQ, in the past reviews, the board suggested to add the parameter, so, what it would be the best practice here?

Reply:
I found that I hit an issue last year when changing a swagger path. The [1005 rule ](https://github.com/Azure/openapi-diff/blob/master/docs/rules/1005.md) I was breaking stated

> Description: Checks whether an existing path is removed from the previous specification.
> Cause: This is considered a breaking change. This change requires new api-version.

and I reached out to Joel Hendrix about why my change, which contained a new API version, appeared to be getting this error.

Joel told me this is because the API version wasn't actually specified in my ApiVersionParameter param (in either swagger) so the comparison thinks they're the same version and I should add it to the enum like this https://github.com/Azure/azure-rest-api-specs/blob/master/specification/cognitiveservices/data-plane/TextAnalytics/stable/v3.1/TextAnalytics.json#L2667

This resulted in me adding the enum to the ApiVersionParameter for the previous swaggers: https://github.com/Azure/azure-rest-api-specs/pull/14955/files

However not it seems that I shouldn't be including this enum?

CC: @DominikMe

Reply:
After discussion, it was decided that this enum should be removed